### PR TITLE
Data entry component

### LINKED ISSUE #13: Create data entry component
### Goal: Create a component for users to enter data!

**Branch Name:** `data-entry-component`

### What You Have To Do
- Create a data entry component:
- For now, that's just going to be 5 text boxes (with labels) for the following data-types (excluding ID, that's auto generated):
image
- Create a submission button that displays the entered data in some way (as an alert if you want)!

### Task Tracker:
- [x] Create branch
- [x] Create data entry component
- [x] Create submission button
- [x] Test your component
- [x] Create a PR!

### PR DESCRIPTION
**Pull request by hanahhleee**
## Overview

Closes #13 : We created a data entry component with the fields date (Date object), fog net id (string), cluster id (string), model name (string), and water collected (number). Upon clicking on the submit button, it displays the entered data in an alert.  

## Changes Made

The date field is auto-filled with today's date but can be changed to whatever date. This date is validated to make sure that the date is a real date. Water collected is validated to be a number, specifically a float, but note that if the inputted number is a decimal that ends in 0 (ex. 15.0), it will interpret the input as an integer.

## Test Coverage

1. checkout branch data-entry-component
2. If submit with empty field, alert that the field is missing
3. If date is not valid, alert that the date is invalid
4. If all fields filled out and valid, alert complete data entry

## Next Steps (delete if not applicable)

The next step would be to store the actual data that is entered into the database. Note that we may need to circle back to this if the lack of type float in js interferes with the data collection.

### CODE REVIEW

File: mobile/components/CustomButton/CustomButton.tsx [hunk 1]
```diff
@@ -12,20 +12,28 @@ import {StyleSheet, Text, Pressable, Alert} from 'react-native';
...
-return(
-  <Pressable onPress={() => Alert.alert('Simple Button pressed!')} style={styles.button1}> 
-    <Text>I'm a button!</Text>
-  </Pressable>
-)};
+
+type ButtonProps = {
+  text: string,
+  handleClick: () => {},
+  css: {},
```
Comment (jfeuerstein) [inline mobile/components/CustomButton/CustomButton.tsx]: Try using a safer type here
```suggestion
  css: object,
```

File: mobile/components/CustomButton.tsx [hunk 2]
```diff
@@ -12,20 +12,28 @@ import {StyleSheet, Text, Pressable, Alert} from 'react-native';
...
-const CustomButton = () => {
-return(
-  <Pressable onPress={() => Alert.alert('Simple Button pressed!')} style={styles.button1}> 
-    <Text>I'm a button!</Text>
-  </Pressable>
-)};
+
+type ButtonProps = {
+  text: string,
+  handleClick: () => {},
```
Comment (jfeuerstein) [inline mobile/components/CustomButton.tsx:18]: Try to make the typing a little more clear
```suggestion
  handleClick: () => object,
```

**Review by jfeuerstein — CHANGES_REQUESTED** [review]
Left a couple of comments (and made a few changes in person with you) -- looks great so far! Once you make these changes and push it should be good to go!

File: mobile/components/CustomButton/DataEntry.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  const [clusterID, setClusterID] = useState("");
+  const [fogNetModel, setFogNetModel] = useState("");
+  const [waterCollected, setWaterCollected] = useState(0.0);
+
+  // Set placeholder date as today's date
+  const currDate = new Date();
+  const month = (currDate.getMonth() + 1 < 10 ? '0' : '') + (currDate.getMonth() + 1);
+  const day = (currDate.getDate() < 10 ? '0' : '') + currDate.getDate();
+  const year = currDate.getFullYear();
+  const [date, setDate]: any = useState(month.toString() + "/" + day.toString() + "/" + year.toString());
```
[the remark this answers is not in the document]
Comment (hanahhleee) [reply mobile/components/CustomButton/DataEntry.tsx]: It doesn't seem to like it when I explicitly type it as a string. I'm thinking that the type of the useState() auto-types it to be a string, but I might be wrong?

File: mobile/App.tsx [hunk 4]
```diff
@@ -3,21 +3,32 @@ import { StyleSheet, View } from 'react-native'
...
 
   return (
-    <View style={styles.container}>
-      <StatusBar style="auto" />
-      <PhoneAuth config={firebaseConfig} auth={auth} />
-    </View >
+    <KeyboardAwareScrollView contentContainerStyle={styles.scrollContainer}>
+      <View style={styles.container}>
+        <StatusBar style="auto" />
+        <PhoneAuth config={firebaseConfig} auth={auth} />
```
[the remark this answers is not in the document]
Comment (hanahhleee) [reply mobile/App.tsx:18]: When I get rid of the auth component, it raises a Render Error that says "Firebase: Need to provide options, when not being deployed to hosting via source", which is why I left it there. Is there a way around this?

**Pull request merged**